# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl
# Commit: ccd96b99aea2
# Full Hash: ccd96b99aea28e8214e585b864d3fb2245bac142
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-06-07 08:31:23
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing failures complaining about fedcm-no-registered-idps.https.html.
#   
#   This reverts commit ba33457e0cbac24a8b7d963c45138c62e92a74ae.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4bfadfc33207 -r ccd96b99aea2 toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl
--- a/toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl	Fri Jun 06 21:07:41 2025 +0300
+++ b/toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl	Fri Jun 06 21:35:18 2025 +0300
@@ -31,6 +31,22 @@
     // Delete the entry for all tuples (rpPrincipal, idpPrincipal, *).
     void disconnect(in nsIPrincipal rpPrincipal, in nsIPrincipal idpPrincipal);
 
+     // Store a "lightweight" identity credential
+     [noscript]
+    void storeIdentityCredential(in IPCIdentityCredentialRef credential);
+
+    // Retrieve all "lightweight" identity credentials stored by these IDPs
+     [noscript]
+    Array<IPCIdentityCredential> getIdentityCredentials(in Array<nsIPrincipal> idpPrincipal);
+
+    // Retrieve all "lightweight" identity credentials stored with this type, for the provided RP
+     [noscript]
+    Array<IPCIdentityCredential> getIdentityCredentialsOfType(in ACString type);
+
+    // Delete a  "lightweight" identity credential
+     [noscript]
+    void deleteIdentityCredential(in IPCIdentityCredentialRef credential);
+
     // Delete all data in this service.
     void clear();
 